### PR TITLE
Add frame rate limiter (fixes #29)

### LINKED ISSUE #29: Framerate limit
Implementing some form of FPS limit would be cool as it is maxing out the GPU (Nvidia control panel on Ubuntu says 100% usage and fans spin up).

### PR DESCRIPTION
**Pull request by bjorn3**

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -220,39 +218,61 @@ fn main() {
...
-            .tick(&mut ui_container, &mut game.renderer, delta, width as f64);
-        ui_container.tick(&mut game.renderer, delta, width as f64, height as f64);
-        game.renderer.tick(&mut game.server.world, delta, width, height);
-
-        window.gl_swap_window();
-
-        for event in events.poll_iter() {
-            handle_window_event(&window, &mut game, &mut ui_container, event)
+        
+        diff = time::Duration::span(||{
```
Comment (Thinkofname) [inline src/main.rs]: `span` is deprecated in the stdlib version of time which I hope to be able to move to in the future once its all stable

File: src/main.rs [hunk 2]
```diff
@@ -220,39 +218,61 @@ fn main() {
...
+
+            window.gl_swap_window();
+
+            for event in events.poll_iter() {
+                handle_window_event(&window, &mut game, &mut ui_container, event)
+            }
+        });
+        
+        //frame rate limiter
+        if frame_rate != 0{
```
Comment (Thinkofname) [inline src/main.rs]: Formatting issue

File: src/main.rs [hunk 3]
```diff
@@ -220,39 +218,61 @@ fn main() {
...
+            let frame_length = (1000.0 as f64) / (frame_rate as f64);
+            let render_time = diff.num_seconds() as f64 * 1000.0 as f64 + (diff.num_milliseconds() as f64);
+            let mut diff = frame_length - render_time;
+            
+            if diff < 0.0{
```
Comment (Thinkofname) [inline src/main.rs]: Formatting issue

File: src/main.rs [hunk 4]
```diff
@@ -220,39 +218,61 @@ fn main() {
...
+                diff = 0.0;
+            }
+            
+            ::std::thread::sleep(::std::time::Duration::from_millis((diff as f32) as u64));
```
Comment (Thinkofname) [inline src/main.rs]: Not a fan of mixing two time libs, we should either switch to the stdlib version or use the older sleep_ms method for now

**Comment by Thinkofname** [comment]
Ideally the framerate to should be changeable on the fly too (as the config value updates in the soon to be added settings menu for it)

File: src/main.rs (hunk 4 above)
Comment (bjorn3) [inline src/main.rs]: The cast is a rest of code I was using.
It be removed

**Comment by bjorn3** [comment]
I will update my code when I am home

**Comment by bjorn3** [comment]
Clos

**Comment by bjorn3** [comment]
Wrong button.
Closed in favor of #42

**Pull request closed without merging**